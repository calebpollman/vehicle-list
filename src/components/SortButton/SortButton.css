.sort-button {
  background-color: var(--color-sort-button-background);
  border: none;
  color: var(--color-sort-bar-text);
  cursor: pointer;
  font-family: var(--font-application);
  font-size: var(--font-size-small);
  font-weight: var(--font-weight-regular);
  height: 100%;
  outline: none;
  transition: all linear 0.3s;
  width: 20%;
}

.sort-button-active {
  font-weight: var(--font-weight-bold);
}

.sort-button:hover {
  background-color: var(--color-sort-button-background-hover);
  color: var(--color-sort-bar-text-hover);
}