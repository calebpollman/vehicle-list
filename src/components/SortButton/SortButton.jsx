import React from 'react';
import PropTypes from 'prop-types';
import './SortButton.css';

const SortButton = ({isActive, sortOption, updateSortSelection}) => {
  let sortClass = 'sort-button';
  if (isActive) sortClass = `${sortClass} ${sortClass}-active`;
  
  return (
    <button 
      className={sortClass}
      onClick={event => updateSortSelection(event, sortOption.toLowerCase())}
    >
      {sortOption}
    </button>
  );
}

SortButton.propTypes = {
  isActive: PropTypes.bool.isRequired,
  sortOption: PropTypes.string.isRequired,
  updateSortSelection: PropTypes.func,
}

export default SortButton;