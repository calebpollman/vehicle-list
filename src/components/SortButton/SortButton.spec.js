import React from 'react';
import {shallow} from 'enzyme';
import SortButton from './SortButton';

describe('SortButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SortButton
        isActive={false}
        sortOption="Make"
      />
    );
  });

  it('should render a <Button />', () => {
    expect(wrapper.find('button').length).toEqual(1);
  });
});