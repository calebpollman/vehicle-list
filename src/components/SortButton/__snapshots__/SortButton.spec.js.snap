// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButton should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SortButton
    isActive={false}
    sortOption="Make"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Make",
      "className": "sort-button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Make",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Make",
        "className": "sort-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Make",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
